packages/jbrowse-plugin-apollo/src/components/ImportFeatures.tsx

```diff
@@ -46,6 +54,64 @@ export function ImportFeatures({
     )
   }
 
+  function handleDeleteFeatures(e: React.ChangeEvent<HTMLInputElement>) {
+    setDeleteFeatures(e.target.checked)
+  }
+
+  // fetch and set features count for selected assembly
+  useEffect(() => {
+    if (assemblyId) {
+      updateFeaturesCount()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [assemblyId])
+
+  function internetAccount() {
+    const apolloInternetAccounts = internetAccounts.filter(
+      (ia: { type: string }) => ia.type === 'ApolloInternetAccount',
+    ) as ApolloInternetAccountModel[]
+
+    if (apolloInternetAccounts && apolloInternetAccounts.length) {
+      return apolloInternetAccounts[0]
+    }
+  }
+
+  function updateFeaturesCount() {
+    const apolloInternetAccount = internetAccount()
+    if (!apolloInternetAccount) {
+      throw new Error('No Apollo internet account found')
+    }
+
+    const { baseURL } = apolloInternetAccount
+    const uri = new URL('/features/count', baseURL)
+    const searchParams = new URLSearchParams({
+      assemblyId,
+    })
+    uri.search = searchParams.toString()
+    const fetch = apolloInternetAccount?.getFetcher({
+      locationType: 'UriLocation',
+      uri: uri.toString(),
+    })
+
+    if (fetch) {
+      // sumbit might get enabled when we change assembly before loading features count
+      setFeaturesCount(-1)
+      setLoading(true)
+      fetch(uri, {
+        method: 'GET',
+      })
+        .then((res) => res.json())
```

Comment:
This should be an `async` function and then use `await` instead of chaining `then`s.